operator/main.go

```diff
@@ -124,6 +124,31 @@ func initEnv() {
 	option.Config.EnableK8sLeasesFallbackDiscovery()
 }
 
+func initK8s(k8sInitDone chan struct{}) {
+	k8s.Configure(
+		option.Config.K8sAPIServer,
+		option.Config.K8sKubeConfigPath,
+		float32(option.Config.K8sClientQPSLimit),
+		option.Config.K8sClientBurst,
+	)
+	if err := k8s.Init(option.Config); err != nil {
+		log.WithError(err).Fatal("Unable to connect to Kubernetes apiserver")
+	}
+
+	close(k8sInitDone)
+
+	// We try to update the Kubernetes version and capabilities. If this fails,
+	// we cannot move forward as we can misinterpret the available Capabilities.
+	// For example, in a case where we got an error when checking for Leases support
+	// we should not assume that support is not available and run the operator
+	// in non HA mode. As this can conflict with other operator instances running
+	// in the cluster, that figured the capabilities correctly and is running in HA
+	// mode.
+	if err := k8sversion.Update(k8s.Client(), option.Config); err != nil {
```

Comment:
Isn't this done by `k8s.Init()` already? I think that in case we cannot update k8s version and capabilities, we will Fatal earlier in line 135.

Reply:
Thanks for pointing this out. According to git blame, that block of code was added by @fristonio. I am going to remove it for now, and await his feedback.

Reply:
Good catch, the changes look good. I think I missed that we already update capabilities in `k8s.Init`. Thanks @mrostecki